tensordict/tensordict.py

```diff
@@ -2562,22 +2562,63 @@ def __torch_function__(
             return NotImplemented
         return TD_HANDLED_FUNCTIONS[func](*args, **kwargs)
 
+    @overload
+    def to(
+        self: T,
+        device: Optional[Union[int, device]] = ...,
+        dtype: Optional[Union[torch.device, str]] = ...,
+        non_blocking: bool = ...,
+    ) -> T:
+        ...
+
+    @overload
+    def to(self: T, dtype: Union[torch.device, str], non_blocking: bool = ...) -> T:
+        ...
+
+    @overload
+    def to(self: T, tensor: Tensor, non_blocking: bool = ...) -> T:
+        ...
+
+    @overload
+    def to(self: T, *, other: T, non_blocking: bool = ...) -> T:
+        ...
+
+    @overload
+    def to(self: T, *, batch_size: torch.Size) -> T:
+        ...
+
     @abc.abstractmethod
-    def to(self, dest: DeviceType | type | torch.Size, **kwargs) -> TensorDictBase:
-        """Maps a TensorDictBase subclass either on a new device or to another TensorDictBase subclass (if permitted).
+    def to(self, *args, **kwargs) -> T:
+        """Maps a TensorDictBase subclass either on another device, dtype or to another TensorDictBase subclass (if permitted).
 
         Casting tensors to a new dtype is not allowed, as tensordicts are not bound to contain a single
         tensor dtype.
 
         Args:
-            dest (device, size or TensorDictBase subclass): destination of the
-                tensordict. If it is a torch.Size object, the batch_size
-                will be updated provided that it is compatible with the
-                stored tensors.
+            device (torch.device, optional): the desired device of the parameters
+                and buffers in this module
+            dtype (torch.dtype, optional): the desired floating point or complex dtype of
+                the parameters and buffers in this module
+            tensor (torch.Tensor, optional): Tensor whose dtype and device are the desired
```

Comment:
So these changes are applied only to parameters and buffers? Not to all tensors contained?

Reply:
it's a copy paste from nn.Module, let me correct that